clang/lib/Sema/SemaSYCL.cpp

```diff
@@ -1246,7 +1245,15 @@ void Sema::ConstructOpenCLKernel(FunctionDecl *KernelCallerFunc,
   assert(TemplateArgs && "No template argument info");
   QualType KernelNameType = TypeName::getFullyQualifiedType(
       TemplateArgs->get(0).getAsType(), getASTContext(), true);
-  std::string Name = constructKernelName(KernelNameType, MC);
+
+  std::string Name;
+  // TODO SYCLIntegrationHeader also computes unique stable name. It should
+  // probably loose this responsibility and only use provided name.
+  if (!getLangOpts().SYCLUnnamedLambda)
+    Name = constructKernelName(KernelNameType, MC);
+  else
+    Name = PredefinedExpr::ComputeName(
```

Comment:
Would a ternary operator be too gross?

```
std::string Name = getLangOpts().SYCLUnnamedLambda ? ...
```

Reply:
I think, if block is more readable. Do you have other opinion?

Reply:
I'm 100% to stick with the block.